### PR TITLE
fix lints from 1.79-1.81

### PR DESCRIPTION
**Pull request by trinity-1686a**
### Description

fix lints from rust/clippy 1.79-1.81

### CODE REVIEW

File: quickwit/quickwit-indexing/Cargo.toml [hunk 1]
```diff
@@ -101,6 +101,7 @@ testsuite = [
   "quickwit-storage/testsuite"
 ]
 vrl = ["dep:vrl", "quickwit-config/vrl"]
+ci-test = []
```
Comment (trinity-1686a) [inline quickwit/quickwit-indexing/Cargo.toml:104]: I don't think this feature is ever used, but some tests depends on it. We might want to enable it somewhere

File: quickwit/quickwit-ingest/src/metrics.rs [hunk 2]
```diff
@@ -28,6 +28,7 @@ pub struct IngestMetrics {
 
     pub replicated_num_bytes_total: IntCounter,
     pub replicated_num_docs_total: IntCounter,
+    #[allow(dead_code)] // this really shouldn't be dead, it needs to be used somewhere
     pub queue_count: IntGauge,
```
Comment (trinity-1686a) [inline quickwit/quickwit-ingest/src/metrics.rs:32]: this ought to be used somewhere

File: quickwit/quickwit-search/Cargo.toml [hunk 3]
```diff
@@ -60,3 +60,4 @@ quickwit-storage = { workspace = true, features = ["testsuite"] }
 
 [features]
 testsuite = []
+ci-test = []
```
Comment (trinity-1686a) [inline quickwit/quickwit-search/Cargo.toml:63]: I don't think this feature is ever used, but some tests depends on it. We might want to enable it somewhere

**Review by rdettai — APPROVED** [review]
we should probably update the Rust version in `rust-toolchain.toml` quickly to prevent the formatting from drifting again

**Pull request merged**